src/Engine/ProtoCore/Utils/ArrayUtils.cs

```diff
@@ -131,8 +131,6 @@ public static ClassNode GetGreatestCommonSubclassForArray(StackValue array, Runt
             var dsArray = runtimeCore.Heap.ToHeapObject<DSArray>(array);
             foreach (var sv in dsArray.Values)
             {
-                if(IsEmpty(sv, runtimeCore)) continue;
```

Comment:
are you able to document what the behavior of this function should be?
`GetTypeExamplesForLayer` ? and add it as a /// comment?

Does this change - sufficiently change that description of the behavior if it had already existed?

Reply:
Yeah I will add the comments for the behaviour of the function. I do not see any description already present. 
From my understanding: If an empty list is passed into a function that takes in an object, we can just skip the empty list completely as it won't needed. But if we are passing the empty list into a function that takes a list, then we cannot skip it because it can match a FEP with empty array-pointer as reduced params and we cannot just skip it then.  @aparajit-pratap should know more

Reply:
@reddyashish While the tests may still pass after this, I think this could have changed the behaviour of function endpoint matching for some cases. To give more context, that line was added by @aparajit-pratap in this PR #9559 , and it seems he was addressing performance improvements. I'm not sure if we still have those tests around, but it would be nice to run them to make sure we did not regress in that.

Reply:
I think the fix that was added in the PR-9559 was mainly to address the failing regression from 2.0. This regression happened after some behavioral change to nested lists. Also, the performance results in that PR show that they were almost same before and after the fix. So i doubt if it was about performance improvement. Either way, I agree it is good to run them again. Will be doing that.